Tevling/Pages/PublicProfile.razor.cs

```diff
@@ -0,0 +1,117 @@
+using Microsoft.Extensions.Localization;
+
+namespace Tevling.Pages;
+
+public partial class PublicProfile : ComponentBase
+{
+    [Inject] private IAuthenticationService AuthenticationService { get; set; } = null!;
+    [Inject] private IStringLocalizer<PublicProfile> Loc { get; set; } = null!;
+
+    [Inject] private IAthleteService AthleteService { get; set; } = null!;
+
+    [Inject] private IBrowserTime BrowserTime { get; set; } = null!;
+    [Inject] private IChallengeService ChallengeService { get; set; } = null!;
+    
+
+
+    [Parameter] public int AthleteToViewId { get; set; }
+    private Athlete Athlete { get; set; } = default!;
+    private Athlete AthleteToView { get; set; } = default!;
+    private string? CreatedTime;
+    private Dictionary<string, (string, string)> ActiveChallenges { get; } = [];
+
+
+    protected override async Task OnInitializedAsync()
+    {
+        Athlete = await AuthenticationService.GetCurrentAthleteAsync();
+        AthleteToView = await AthleteService.GetAthleteByIdAsync(AthleteToViewId) ??
+            throw new InvalidOperationException($"Athlete with ID {AthleteToViewId} not found");
+    }
+
+    protected override async Task OnParametersSetAsync()
+    {
+        AthleteToView = await AthleteService.GetAthleteByIdAsync(AthleteToViewId) ??
+            throw new InvalidOperationException($"Athlete with ID {AthleteToViewId} not found");
```

Comment:
Rather than throwing an exception here, perhaps it would be better to have `AthleteToView` nullable and have the template print a 404 message if it's null? See the `EditChallenge` page for reference.